recipes/leveldb/all/conanfile.py

```diff
@@ -0,0 +1,71 @@
+import os
+from conans import CMake, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+from conans.tools import Version
+
+
+class LevelDBCppConan(ConanFile):
+    name = "leveldb"
+    description = "LevelDB is a fast key-value storage library written at Google that provides an ordered mapping from string keys to string values."
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/google/leveldb"
+    topics = ("conan", "leveldb", "db")
+    license = "MIT"
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake", "cmake_find_package"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False], 
+        "with_snappy": [True, False],
+        "fPIC": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_snappy": True,
```

Comment:
did you by any chance try `with_snappy` true and another snappy installed in your system? I can give this a try as well, but I think I ran into an issue with the wrong library (the system-installed one) getting picked. That's why I added patch sources portion in https://github.com/conan-io/conan-center-index/pull/1542/files#diff-224f12e422e56324bbccab54a890d9b3R57
although it has been a while since I worked on it so I may not be remembering details right. (I had written this a while ago in my repo: https://github.com/jjkoshy/conan-recipes/tree/master/conan-leveldb)

Reply:
So I think I remember this better now - basically, if a build specifies `with_snappy` false but the underlying machine has a pre-installed snappy in a standard library location, then libleveldb will incorporate references to that snappy library. i.e., if a consumer of the package wishes to use leveldb without snappy, there will be link errors. So that's why I think we should ideally patch leveldb's cmakelists.txt if `with_snappy` is false.